samples/logs.js

```diff
@@ -16,48 +16,51 @@
 
 async function writeLogEntry(logName) {
   // [START logging_write_log_entry]
-  // Imports the Google Cloud client library
   const {Logging} = require('@google-cloud/logging');
-
-  // Creates a client
   const logging = new Logging();
 
   /**
-   * TODO(developer): Uncomment the following line to run the code.
+   * TODO(developer): Uncomment the following line and replace with your values.
    */
-  // const logName = 'Name of the log to write to, e.g. my-log';
-
+      // const logName = 'my-log';
   const log = logging.log(logName);
 
-  // Modify this resource to match a resource in your project
-  // See
-  // https://cloud.google.com/logging/docs/api/ref_v2beta1/rest/v2beta1/MonitoredResource
-  const resource = {
-    // This example targets the "global" resource for simplicity
-    type: 'global',
-  };
-
   // A text log entry
-  const entry = log.entry({resource}, 'Hello, world!');
-
-  // A structured log entry
-  const secondEntry = log.entry(
-    {resource: resource},
-    {
-      name: 'King Arthur',
-      quest: 'Find the Holy Grail',
-      favorite_color: 'Blue',
+  const text_entry = log.entry('Hello world!');
+
+  // A json log entry with additional context
+  const metadata = {
+    severity: 'WARNING',
+    labels: {
+      foo: "bar"
+    },
+    // A default log resource is added for some GCP environments
+    // This log resource can be overwritten per spec:
+    // https://cloud.google.com/logging/docs/reference/v2/rest/v2/MonitoredResource
+    resource: {
+      type: 'global'
     }
-  );
+  }
 
-  async function writeLogEntry() {
-    // Save the two log entries. You can write entries one at a time, but it is
-    // best to write multiple entires together in a batch.
-    await log.write([entry, secondEntry]);
-    console.log(`Wrote to ${logName}`);
+  const message = {
+    name: 'King Arthur',
+    quest: 'Find the Holy Grail',
+    favorite_color: 'Blue',
   }
-  writeLogEntry();
+
+  const json_Entry = log.entry(metadata, message);
+
+  // Synchronously write the log entry
+  await log.write(text_entry);
```

Comment:
As this is written right now, the user will see `await` calls without a wrapping async function.  This will lead to copy/paste code that doesn't work sadly.  This needs an outer `main` function that holds an async function that does an await.  I know this is kind of terrible.   For a good example:
https://github.com/googleapis/nodejs-bigquery/blob/master/samples/addColumnLoadAppend.js

Reply:
Ah yes we want it to be copy&paste runnable.

@JustinBeckwith should I refactor all node tests to have an outer main (might take some time). 
Or is it ok if I wrap the await call inside an async function like the [current state of the example](https://cloud.google.com/logging/docs/samples/logging-write-log-entry#logging_write_log_entry-nodejs)? 

```js
...
async function writeLogEntry() {
  // Save the two log entries. You can write entries one at a time, but it is
  // best to write multiple entires together in a batch.
  await log.write([entry, secondEntry]);
  console.log(`Wrote to ${logName}`);
}
writeLogEntry();
``

Reply:
Shouldn't have to refactor the tests - just the samples themselves.  What you have written there is perfect.  That whole thing though ends up in it's own `main` function usually.

Reply:
👍  updated . thanks for the feedback!